packages/toolpad-app/src/toolpad/AppEditor/PageEditor/RenderPanel/RenderOverlay.tsx

```diff
@@ -1011,14 +1013,20 @@ export default function RenderOverlay({ canvasHostRef }: RenderOverlayProps) {
         }
       }
 
-      api.dragEnd();
-
       if (selection) {
         deleteOrphanedLayoutComponents(draggedNode, dragOverNodeId);
       }
 
+      api.dragEnd();
+
       if (newNode) {
         api.select(newNode.id);
+
+        // Refocus on overlay so that keyboard events can keep being caught by it
+        const overlayElement = overlayRef.current;
+        if (overlayElement) {
+          overlayElement.focus();
+        }
```

Comment:
Do we need the defensive check here? I would assume React guarantees it's defined, hence:
```suggestion
        overlayRef.current!.focus();
```

would be better, so we catch regressions at runtime

Reply:
In [some other places](https://github.com/mui/mui-toolpad/blob/a1794f5c61e160c3df82a9fabe77728d3181fab4/packages/toolpad-app/src/components/MonacoEditor.tsx#L279) we do:

```suggestion
        invariant(overlayRef.current, 'Ref not bound');
        overlayRef.current.focus();
```

This communicates the intent a bit clearer and avoids usage of `!`

Reply:
good points, thanks! i used an `invariant` check here and also in a couple other places where it seemed right to use it